### PR TITLE
Enable Auth0 Login for All Preview URLs

### PR DESCRIPTION
**Pull request by sagararyal**
Tests need to be made with VERCEL URLs.

If you get an error with URL eg: https://planet-webapp-git-feature-auth0-url-test.planetapp.vercel.app/

**Please follow the URL you get with Error.**

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -27,6 +27,10 @@ const scheme = process.env.SCHEME === 'http' || process.env.SCHEME === 'https'
   ? process.env.SCHEME
   : 'https';
 
+const nextauthUrl = process.env.NEXTAUTH_URL === '' || process.env.NEXTAUTH_URL === 'null'
```
Comment (norbertschuler) [inline next.config.js:30]: Are you sure about testing against a string value of 'null' - normally someone is testing again the value null (not as a string)? I lately changed all this empty string plus null checks into simple checks just if the variable contains a value like this:

```
process.env.NEXTAUTH_URL  ?  `${scheme}://${process.env.NEXTAUTH_URL}` : process.env.VERCEL_URL ;
```

**Comment by norbertschuler** [comment]
> **Please follow the URL you get with Error.**

Login on https://planet-webapp-760gjyryr.vercel.app/ works for me.

**Review by norbertschuler — COMMENTED** [review]
Login on https://planet-webapp-760gjyryr.vercel.app/ works for me. Is that the intention of this PR?

**Comment by harshvitra** [comment]
I was thinking that we could also use document url and extract the domain from there. 

The only challenge will be to allow all such URLs in the Auth0.com dashboard

**Comment by harshvitra** [comment]
Screenshot 2020-11-06 at 2 15 26 PM

**Comment by harshvitra** [comment]
https://planet-webapp-git-feature-auth0-url-test.planetapp.vercel.app/

Takes me to https://planet-webapp-760gjyryr.vercel.app/

and then it works

**Comment by norbertschuler** [comment]
> This way developers working locally will not need to have https. @norbertschuler

This PR works well for me on localhost.

**Comment by norbertschuler** [comment]
Logging out on https://planet-webapp-760gjyryr.vercel.app/ produces an Auth0 error.

**Comment by sagararyal** [comment]
> Logging out on https://planet-webapp-760gjyryr.vercel.app/ produces an Auth0 error.

@harshvitra

**Pull request merged**